core/src/main/java/org/jdbi/v3/Foreman.java

```diff
@@ -34,8 +37,15 @@
         this.factories.addAll(factories);
     }
 
-    Argument waffle(Class<?> expectedType, Object it, StatementContext ctx)
+    Argument waffle(Type expectedType, Object it, StatementContext ctx)
     {
+        return waffle(new TypeResolver().resolve(expectedType), it, ctx);
+    }
+
+    Argument waffle(ResolvedType expectedType, Object it, StatementContext ctx)
+    {
+        ResolvedType objectType = new TypeResolver().resolve(Object.class);
```

Comment:
I don't know if there is a large efficiency gain or not.  The documentation encourages reuse of the `TypeResolver` instance so I was just blindly recommending doing so.  But yeah I really meant the `ResolvedType` here, not the `TypeResolver`.